src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/collator/Collator.java

```diff
@@ -0,0 +1,125 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Collator collects and prunes the candidate decisions from each decider so that their impacts are
+ * aligned
+ *
+ * <p>Decisions can increase or decrease pressure on different key resources on an Elasticearch
+ * node. This is encapsulated in each Action via the {@link ImpactVector}. Since each decider
+ * independently evaluates its decision, it is possible to have conflicting ImpactVectors from
+ * actions across deciders.
+ *
+ * <p>The collator prunes them to ensure we only take actions that either increase, or decrease
+ * pressure on a particular node. To resolve conflicts, we prefer stability over performance.
+ */
+public class Collator extends Decider {
+
+  public static final String NAME = "collator";
+
+  /* Deciders can choose to publish decisions at different frequencies based on the
+   * type of resources monitored and rca signals. The collator should however, not introduce any
+   * unnecessary delays. As soon as a decision is taken, it should be evaluated and published downstream.
+   */
+  private static final int collatorFrequency = 1; // Measured in terms of number of evaluationIntervalPeriods
+
+  private final List<Decider> deciders;
+
+  private final ActionGrouper actionGrouper;
+
+  public Collator(long evalIntervalSeconds, Decider... deciders) {
+    this(evalIntervalSeconds, new SingleNodeImpactActionGrouper(), deciders);
+  }
+
+  public Collator(long evalIntervalSeconds, ActionGrouper actionGrouper, Decider... deciders) {
+    super(evalIntervalSeconds, collatorFrequency);
+    this.deciders = Arrays.asList(deciders);
+    this.actionGrouper = actionGrouper;
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public Decision operate() {
+    List<Action> proposedActions = getAllProposedActions();
+    Map<NodeKey, List<Action>> actionsByNode = actionGrouper.groupByNodeId(proposedActions);
+    List<Action> prunedActions = new ArrayList<>();
+    actionsByNode.forEach((k, v) -> prunedActions.addAll(polarize(k, v)));
+
+    Decision finalDecision = new Decision(System.currentTimeMillis(), NAME);
+    finalDecision.addAllActions(prunedActions);
+    return finalDecision;
+  }
+
+  @NonNull private List<Action> getAllProposedActions() {
+    final List<Action> proposedActions = new ArrayList<>();
+    if (deciders != null) {
+      for (final Decider decider : deciders) {
+        List<Decision> decisions = decider.getFlowUnits();
+        if (decisions != null) {
+          decisions.forEach(decision -> proposedActions.addAll(decision.getActions()));
+        }
+      }
+    }
+    return proposedActions;
+  }
+
+  private List<Action> polarize(final NodeKey nodeKey, List<Action> actions) {
```

Comment:
maybe we can make actions final as well ?

Reply:
done. Took care of the other parameters as well.